src/deepqmc/sampling.py

```diff
@@ -365,6 +365,38 @@ def propagate_all(self):
         self.restart()
 
 
+def sort_nucleus_indices(idxs, coords):
+    # this heuristic takes a batch of nuclear indices for placing electrons
+    # and sorts them such that the local electronic spin is minimized
+    bs, n_electrons = idxs.shape
+    idx_new = []
+    idx_map = torch.arange(n_electrons).repeat(bs, 1)
+    idx_i = torch.randint(0, n_electrons, (bs,))
```

Comment:
It seems that this is the only source of randomness. Does that mean that there will be only `n_electrons` different assignments of electrons to nuclei? Previously, the electrons were randomly permuted. I think you could leave that permutation in, apply your heuristic afterwards, and it would still work, or?

Reply:
Yes, this would be possible, but if you consider that same-spin electrons are identical and we want to avoid those random permutations which place (more than one) unpaired electrons within every subsystem, it turns out that this is the only randomness that remains. In fact permuting the indices before applying the heuritsic would have no effect.

Reply:
Hm, I would think that with `n_electrons`, I have `n_electrons / 2` pairs, which I can place among the `n_nuclei` however I wish, so that should be `(n_electrons / 2) ** n_nuclei` possibilities at least.

Reply:
Hm, I would think that with `n_electrons`, I have `n_electrons / 2` pairs, which I can place among the `n_nuclei` however I wish, as long as I satisfy the atomic charges `charge[m]` so that should be something like `(n_electrons / 2)! / prod_m (charge[m] / 2)!`

Reply:
But why would you prefere one specific pair over another?  I see how the indices would be different, but the architecture is constructed to be equivariant under the exchange of same spin electrons and hence does not explicitly percieve this permutation.

Reply:
Because as a user of the code, I wouldn't assume there will be any correlations in how the electrons are distributed. So if I'm, say, debugging some problem, I might run into this, and think that it's an actual problem somewhere.

Reply:
I see. The new version mixes the indices accordingly.